src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -640,7 +640,10 @@ public CompilationInfo(Compilation compilation, bool hasSuccessfullyLoaded)
                                 if (metadataReference != null)
                                 {
                                     newReferences.Add(metadataReference);
-                                    metadataReferenceToProjectId.Add(metadataReference, projectReference.ProjectId);
+                                    if (!metadataReferenceToProjectId.ContainsKey(metadataReference))
```

Comment:
If we are going to go ahead with this fix instead of fixing the root problem of duplicate project references, I think we should have an assert here that if there is a duplicate, the Id in the map matches projectReference.ProjectId. If that's _not_ the case then something is really screwed up here and we won't record the right things in the UpdateCompilationWithNewReferencesAndRecordAssemblySymbols call below.